src/probe_modules/module_icmp_echo_time.c

```diff
@@ -35,184 +35,193 @@ struct icmp_payload_for_rtt {
-    struct icmp *icmp_header = (struct icmp*)(&ip_header[1]);
-    struct icmp_payload_for_rtt *payload = (struct icmp_payload_for_rtt*)(((char *)icmp_header) + 8);
-    uint16_t icmp_idnum = validation[2] & 0xFFFF;
-    struct timeval tv;
+	struct ether_header *eth_header = (struct ether_header *) buf;
+	struct ip *ip_header = (struct ip *)(&eth_header[1]);
+	struct icmp *icmp_header = (struct icmp*)(&ip_header[1]);
+	struct icmp_payload_for_rtt *payload =
+			(struct icmp_payload_for_rtt*)(((char *)icmp_header) + 8);
 
-    ip_header->ip_src.s_addr = src_ip;
-    ip_header->ip_dst.s_addr = dst_ip;
+	uint16_t icmp_idnum = validation[1] & 0xFFFF;
+	uint16_t icmp_seqnum = validation[2] & 0xFFFF;
+	struct timeval tv;
 
-    icmp_header->icmp_id = icmp_idnum;
+	ip_header->ip_src.s_addr = src_ip;
+	ip_header->ip_dst.s_addr = dst_ip;
 
-    gettimeofday(&tv, NULL);
-    payload->sent_tv_sec = tv.tv_sec;
-    payload->sent_tv_usec = tv.tv_usec;
-    payload->dst = dst_ip;
 
-    icmp_header->icmp_cksum = 0;
-    icmp_header->icmp_cksum = icmp_checksum((unsigned short *) icmp_header);
+	icmp_header->icmp_id = icmp_idnum;
+	icmp_header->icmp_seq = icmp_seqnum;
 
-    ip_header->ip_sum = 0;
-    ip_header->ip_sum = zmap_ip_checksum((unsigned short *) ip_header);
+	gettimeofday(&tv, NULL);
+	payload->sent_tv_sec = tv.tv_sec;
+	payload->sent_tv_usec = tv.tv_usec;
+	payload->dst = dst_ip;
 
-    return EXIT_SUCCESS;
+	icmp_header->icmp_cksum = 0;
+	icmp_header->icmp_cksum = icmp_checksum((unsigned short *) icmp_header);
+
+	ip_header->ip_sum = 0;
+	ip_header->ip_sum = zmap_ip_checksum((unsigned short *) ip_header);
+
+	return EXIT_SUCCESS;
 }
 
 static void icmp_echo_print_packet(FILE *fp, void* packet)
 {
-    struct ether_header *ethh = (struct ether_header *) packet;
-    struct ip *iph = (struct ip *) &ethh[1];
-    struct icmp *icmp_header = (struct icmp*) (&iph[1]);
-
-    fprintf(fp, "icmp { type: %u | code: %u "
-            "| checksum: %#04X | id: %u | seq: %u }\n",
-            icmp_header->icmp_type,
-            icmp_header->icmp_code,
-            ntohs(icmp_header->icmp_cksum),
-            ntohs(icmp_header->icmp_id),
-            ntohs(icmp_header->icmp_seq));
-    fprintf_ip_header(fp, iph);
-    fprintf_eth_header(fp, ethh);
-    fprintf(fp, "------------------------------------------------------\n");
+	struct ether_header *ethh = (struct ether_header *) packet;
+	struct ip *iph = (struct ip *) &ethh[1];
+	struct icmp *icmp_header = (struct icmp*) (&iph[1]);
+
+	fprintf(fp, "icmp { type: %u | code: %u "
+			"| checksum: %#04X | id: %u | seq: %u }\n",
+			icmp_header->icmp_type,
+			icmp_header->icmp_code,
+			ntohs(icmp_header->icmp_cksum),
+			ntohs(icmp_header->icmp_id),
+			ntohs(icmp_header->icmp_seq));
+	fprintf_ip_header(fp, iph);
+	fprintf_eth_header(fp, ethh);
+	fprintf(fp, "------------------------------------------------------\n");
 }
 
 static int icmp_validate_packet(const struct ip *ip_hdr,
```

Comment:
This problem exists across other modules as well. In our copious amounts of spare time we should try to unify and standardize how this is done.